pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1589,6 +1599,59 @@ type EndpointAddress struct {
 	Hostname string `json:"hostname,omitempty" protobuf:"bytes,3,opt,name=hostname"`
 }
 
+// ProtocolSpec represents a Ceph Object Store protocol specification
+type ProtocolSpec struct {
+	// The spec for S3
+	// +optional
+	// +nullable
+	S3 *S3Spec `json:"s3,omitempty"`
+
+	// The spec for Swift
+	// +optional
+	// +nullable
+	Swift *SwiftSpec `json:"swift"`
+}
+
+// S3Spec represents Ceph Object Store specification for the S3 API
+type S3Spec struct {
+	Enabled         *bool `json:"enabled,omitempty"`
+	AuthUseKeystone *bool `json:"authUseKeystone,omitempty"`
+}
+
+// SwiftSpec represents Ceph Object Store specification for the Swift API
+type SwiftSpec struct {
+	AccountInUrl      *bool   `json:"accountInUrl,omitempty"`
+	UrlPrefix         *string `json:"urlPrefix,omitempty"`
+	VersioningEnabled *bool   `json:"versioningEnabled,omitempty"`
+}
+
+// AuthSpec represents the authentication protocol configuration of a Ceph Object Store Gateway
+type AuthSpec struct {
+	// +optional
+	// +nullable
+	Keystone *KeystoneSpec `json:"keystone,omitempty"`
+}
+
+// KeystoneSpec represents the Keystone authentication configuration of a Ceph Object Store Gateway
+type KeystoneSpec struct {
+	Url                   string                `json:"url"`
+	ServiceUserSecretName string                `json:"serviceUserSecretName"`
+	AcceptedRoles         []string              `json:"acceptedRoles"`
+	ImplicitTenants       ImplicitTenantSetting `json:"implicitTenants,omitempty"`
+	TokenCacheSize        *int                  `json:"tokenCacheSize,omitempty"`
+	RevocationInterval    *int                  `json:"revocationInterval,omitempty"`
```

Comment:
Please add godoc comments to all fields, and mark pointer types as nullable.

Reply:
Addressed in 556765ceea